src/goto-programs/goto_contractor.cpp

```diff
@@ -186,8 +205,13 @@ goto_contractort::create_constraint_from_expr2t(irep_container<expr2t> expr)
 {
   ibex::NumConstraint *c;
   if(
-    is_arith_expr(expr) || is_constant_number(expr) || is_symbol2t(expr) ||
-    is_notequal2t(expr) || is_equality2t(expr))
+    is_arith_expr(get_base_object(expr)) ||
+    is_constant_number(get_base_object(expr)) ||
+    is_symbol2t(get_base_object(expr)) ||
+    is_notequal2t(get_base_object(expr)) ||
+    is_equality2t(get_base_object(expr)) || is_not2t(get_base_object(expr)) ||
+    is_modulus2t(get_base_object(expr)) || is_or2t(get_base_object(expr)) ||
+    is_and2t(get_base_object(expr)))
   {
     log_status("Expression is complex, skipping this assert");
```

Comment:
It should be helpful to print which functions were skipped, you can get the expr name by calling `get_expr_id`.

Also, this looks more like a debug message than a status one, don't you think? I expect to see several of these messages being printed in a row when verifying a program.

Reply:
I agree, changed it to include which expression id and changed the type to debug message.